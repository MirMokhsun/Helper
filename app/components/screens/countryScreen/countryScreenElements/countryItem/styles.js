import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    countryContainer: {
        flexDirection: 'row',
        width: "100%",
        justifyContent: 'space-between',
        height: 50,
        backgroundColor: '#E5E5E5',
        alignItems: 'center',
        borderBottomWidth: 0.5,
        borderColor: "#5A6772",
    },
    flagnadname: {
        flexDirection: 'row',
        alignItems: 'center',
        width: "75%",
        marginRight: 5,
        marginLeft: 5,
    },
    wrapper: {
        flex: 1,
        width: "100%",
        height: "100%",
        backgroundColor: '#ffffff',
    },
    text: {
        fontSize: 14,
        color: 'white',
        fontFamily: 'Roboto-Light',
        color: '#1459D2',
        marginRight: 15,
        marginLeft: 5,
    },
    imagewrapper: {
        height: 30,
        width: 30,
        marginRight: 5,
        marginLeft: 10,
    },
    image: {
        height: 30,
        width: 30,
    },
});