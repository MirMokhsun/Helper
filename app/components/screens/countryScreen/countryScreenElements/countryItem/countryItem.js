import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { styles } from './styles';
import { withNavigation } from 'react-navigation';

export class CountryItem extends Component {
    constructor(props){
        super(props)
    }

    setCountry = () => {
        const { setPhcode, setRegistrationCountry, navigation, country, countryKey } = this.props;
        setPhcode(country.phcode);
        setRegistrationCountry({ isValid: true, value: country.countryName, countryKey });
        navigation.goBack();
    };

    render() {
        const { country, accessibilityLabel, testID, flag } = this.props;
        return (
            <TouchableOpacity onPress={this.setCountry} accessibilityLabel={accessibilityLabel} testID={testID}>
                <View style={styles.countryContainer}>
                    <View style={styles.flagnadname}>
                        <View style={styles.imagewrapper}>
                            {flag}
                        </View>
                        <Text style={styles.text} numberOfLines={1}>{country.countryName}</Text>
                    </View>
                    <Text style={styles.text} >+{country.phcode}</Text>
                </View>
            </TouchableOpacity>
        )
    }
};

export default withNavigation(CountryItem);