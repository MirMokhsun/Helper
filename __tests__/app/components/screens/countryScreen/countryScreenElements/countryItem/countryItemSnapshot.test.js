import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CountryItem } from '../../../../../../../app/components/screens/countryScreen/countryScreenElements/countryItem/countryItem';

describe('CountryItem', () => {
    it('create snapshot CountryItem', () => {
        const renderer = new ShallowRenderer()
        let countryItem = renderer.render(<CountryItem
            key={'key'}
            setRegistrationCountry={() => { }}
            setPhcode={() => { }}
            country={''}
            flag={''}
            testID={'ID'}
            countryKey={'key'}
            accessibilityLabel={'ID'}
        />);
        expect(countryItem).toMatchSnapshot();
    });
});