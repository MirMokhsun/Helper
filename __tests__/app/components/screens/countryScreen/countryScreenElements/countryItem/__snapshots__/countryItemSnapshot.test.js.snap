// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryItem create snapshot CountryItem 1`] = `
<TouchableOpacity
  accessibilityLabel="ID"
  activeOpacity={0.2}
  onPress={[Function]}
  testID="ID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E5E5E5",
        "borderBottomWidth": 0.5,
        "borderColor": "#5A6772",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 5,
          "marginRight": 5,
          "width": "75%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "marginLeft": 10,
            "marginRight": 5,
            "width": 30,
          }
        }
      >
        
      </View>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#1459D2",
            "fontFamily": "Roboto-Light",
            "fontSize": 14,
            "marginLeft": 5,
            "marginRight": 15,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#1459D2",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "marginLeft": 5,
          "marginRight": 15,
        }
      }
    >
      +
    </Text>
  </View>
</TouchableOpacity>
`;
